pilot/cmd/pilot-agent/main.go

```diff
@@ -185,6 +187,7 @@ func init() {
 		"Go template bootstrap config")
 	proxyCmd.PersistentFlags().StringVar(&outlierLogPath, "outlierLogPath", "",
 		"The log path for outlier detection")
+	proxyCmd.PersistentFlags().BoolVar(&noEnvoy, "no-proxy", false, "disables envoy proxy")
```

Comment:
Let's avoid adding more flags - use env variables instead ( we need to get rid of the other flags as well, and merge envs with the ProxyConfig.meta  ).